polkadot/runtime/common/src/slots/mod.rs

```diff
@@ -267,11 +312,17 @@ impl<T: Config> Pallet<T> {
 				if let Some(ended_lease) = maybe_ended_lease {
 					// Then we need to get the new amount that should continue to be held on
 					// deposit for the parachain.
-					let now_held = Self::deposit_held(para, &ended_lease.0);
+					let required_hold = Self::deposit_held(para, &ended_lease.0);
 
 					// If this is less than what we were holding for this leaser's now-ended lease,
 					// then unreserve it.
-					if let Some(rebate) = ended_lease.1.checked_sub(&now_held) {
+					let current_hold = ended_lease.1;
+					// Fixme(ank4n): check reserved balance from state instead of deposit. We might
```

Comment:
Fungibles would fix this but probably shouldn't migrate in this PR

Reply:
Reworking on migrating slots pallet to fungibles [here #1794](https://github.com/paritytech/polkadot-sdk/pull/1794)